### PR TITLE
Secure internal communication with mothership and component-reconcilers

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- allow `mothership-reconciler` to reach `component-reconcilers`
- allow `component-reconcilers` and the KEB to reach the `mothership-reconciler`
- allow KEB to reach the `mothership-reconciler`

**Related issue(s)**

https://github.com/kyma-incubator/reconciler/issues/53

### CODE REVIEW

File: resources/kcp/charts/component-reconcilers/templates/authorization.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{{ $global := . }}
+{{- range $component := .Values.global.components }}
+{{ $subchart := printf "%s-component-reconciler" $component }}
+{{ if not (hasKey $global.Subcharts $subchart) }}
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
```
Comment (Tomasz-Smelcerz-SAP) [inline resources/kcp/charts/component-reconcilers/templates/authorization.yaml:6]: I think you also need RequestAuthentication object defined for the same workload?

Comment (pPrecel) [reply resources/kcp/charts/component-reconcilers/templates/authorization.yaml:6]: Yes, I missed it. Thanks!

Comment (pPrecel) [reply resources/kcp/charts/component-reconcilers/templates/authorization.yaml:6]: Should I use the `RequestAuthentication` if I'm not using JWT? Principals are defined a few lines under ([here](https://github.com/kyma-project/control-plane/pull/1024/files#diff-af952c9ab6230be60c72a211a4ad7ea06f0d38fcb92854df1dc19f45fe9a2cc9R19))

Comment (Tomasz-Smelcerz-SAP) [reply resources/kcp/charts/component-reconcilers/templates/authorization.yaml:6]: 👍

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Review by dbadura — APPROVED** [review]

**Review by PK85 — APPROVED** [review]

**Pull request merged**